Wrapping/Generators/Python/itk/support/helpers.py

```diff
@@ -173,3 +176,123 @@ def image_filter_wrapper(*args, **kwargs):
             return image_filter(*args, **kwargs)
 
     return image_filter_wrapper
+
+
+def wasm_type_from_image_type(itkimage):  # noqa: C901
+    import itk
+
+    component = itk.template(itkimage)[1][0]
+    if component == itk.UL:
+        if os.name == 'nt':
+            return 'uint32_t', 1
+        else:
+            return 'uint64_t', 1
+    mangle = None
+    pixelType = 1
+    if component == itk.SL:
+        if os.name == 'nt':
+            return 'int32_t', 1,
+        else:
+            return 'int64_t', 1,
+    if component in (itk.SC, itk.UC, itk.SS, itk.US, itk.SI, itk.UI, itk.F,
+            itk.D, itk.B):
+        mangle = component
+    elif component in [i[1] for i in itk.Vector.items()]:
+        mangle = itk.template(component)[1][0]
+        pixelType = 5
+    elif component == itk.complex[itk.F]:
+        # complex float
+        return 'float', 10
+    elif component == itk.complex[itk.D]:
+        # complex float
+        return 'double', 10
+    elif component in [i[1] for i in itk.CovariantVector.items()]:
+        # CovariantVector
+        mangle = itk.template(component)[1][0]
+        pixelType = 7
+    elif component in [i[1] for i in itk.Offset.items()]:
+        # Offset
+        return 'int64_t', 4
+    elif component in [i[1] for i in itk.FixedArray.items()]:
+        # FixedArray
+        mangle = itk.template(component)[1][0]
+        pixelType = 11
+    elif component in [i[1] for i in itk.RGBAPixel.items()]:
+        # RGBA
+        mangle = itk.template(component)[1][0]
+        pixelType = 3
+    elif component in [i[1] for i in itk.RGBPixel.items()]:
+        # RGB
+        mangle = itk.template(component)[1][0]
+        pixelType = 2
+    elif component in [i[1] for i in itk.SymmetricSecondRankTensor.items()]:
+        # SymmetricSecondRankTensor
+        mangle = itk.template(component)[1][0]
+        pixelType = 8
+    else:
+        raise RuntimeError('Unrecognized component type: {0}'.format(str(component)))
+    _python_to_js = {
+        itk.SC: 'int8_t',
+        itk.UC: 'uint8_t',
+        itk.SS: 'int16_t',
+        itk.US: 'uint16_t',
+        itk.SI: 'int32_t',
+        itk.UI: 'uint32_t',
```

Comment:
We could add the `long`, `long long` support here as in `_long_type()` in `image_type_from_wasm_type`.

Reply:
This comment suggests that maybe [this](https://github.com/InsightSoftwareConsortium/ITK/pull/2829#discussion_r746316216) was not quite what you had in mind? 

The `_long_type()` function behaves differently on posix vs windows machines, do we need to treat them differently here too? I'm sorry, I don't have a Windows machine handy, or I'd be able to test this out for myself.

Reply:
Yes, on Visual Studio and 64-bit Windows, long is still 32-bits. GCC and clang have long as 64-bits on 64-bit machines, so we do need to treat long differently. I saw an `if os.nt` as a distinguisher already used in the code to treat long differently.

Reply:
This size ambiguity of `long` is a reason why we use the explicit type names in the WebAssembly serialization interface. Recently NumPy has moved to prefer more explicit use of primative type sizes, e.g. `np.float64`.